content/en/docs/reference/access-authn-authz/certificate-signing-requests.md

```diff
@@ -0,0 +1,207 @@
+
+## Authorization
+
+To allow creating a CertificateSigningRequest and retrieving any CertificateSigningRequest:
+* create certificates.k8s.io certificatesigningrequests
+* get certificates.k8s.io certificatesigningrequests
+
+To allow approving a CertificateSigningRequest:
+* update certificates.k8s.io certificatesigningrequests/approval
+* approve certificates.k8s.io signers <signerNameDomain>/<signerNamePath> or
+  approve certificates.k8s.io signers <signerNameDomain>/*
+
+To allow signing a CertificateSigningRequest:
+* update certificates.k8s.io certificatesigningrequests/status
+* sign certificates.k8s.io signers <signerNameDomain>/<signerNamePath> or
+  sign certificates.k8s.io signers <signerNameDomain>/*
+
+## Approval/Rejection
+
+### kube-controller-manager auto-approval
+
+The kube-controller-manager ships with an in-built
+[approver](https://github.com/kubernetes/kubernetes/blob/32ec6c212ec9415f604ffc1f4c1f29b782968ff1/pkg/controller/certificates/approver/sarapprove.go)
+for Kubelet TLS Bootstrap that delegates various permissions on CSRs for node
+credentials to authorization. It does this by posting subject access reviews to
+the API server. It punts on TLS certificates for server authentication of the
+Kubelet API because verifying IP SANs for Kubelets in a generic way poses
+challenges.
+
+The GCP controller manager replaces the in-built approver with an
+[approver](https://github.com/kubernetes/cloud-provider-gcp/blob/08fa1e3260ffb267682762e24ba93692000e3be8/cmd/gcp-controller-manager/csr_approver.go)
+that handles TLS certificates for server authentication and support for
+verification of CSRs with proof of control of GCE vTPM.
+
+An external project, [kapprover](https://github.com/coreos/kapprover), does
+policy based approval of kubelet CSRs.
+
+### kubectl-based approver
+
+A Kubernetes administrator (with appropriate permissions) can manually approve
+(or deny) Certificate Signing Requests by using the `kubectl certificate
+approve` and `kubectl certificate deny` commands.
+
+To approve a CSR with kubectl:
+
+```
+kubectl certificate approve [NAME OF CSR]
+```
+
+Likewise, to deny a CSR:
+
+```
+kubectl certificate deny [NAME OF CSR]
+```
+
+### API-based approver
+
+Users of the REST API can approve CSRs by submitting an UPDATE request to the `approval`
+subresource of the CSR to be approved.
+
+As part of this request, the `Approved` or `Denied` status condition should be set to the
+desired state.
+
+## Signing
+
+### kube-controller-manager signer
+
+The kube-controller-manager ships with an in-built
+[signer](https://github.com/kubernetes/kubernetes/blob/32ec6c212ec9415f604ffc1f4c1f29b782968ff1/pkg/controller/certificates/signer/cfssl_signer.go)
+that signs all approved certificates with a local key. This key is generally
+configured to be the key of the cluster's root certificate authority, but is not
+required to be.
```

Comment:
that signs certificates that have the kubernetes.io signerNames defined above (kube-controller-manager versions prior to the introduction of `signerName` in v1.18 would sign all approved CSRs).

Reply:
Drop `This key is generally...`